File: rust/src/models/resolved_entitlement_list_response.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::resolved_entitlement::ResolvedEntitlement;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct ResolvedEntitlementListResponse {
    pub data: Vec<ResolvedEntitlement>,
}

impl ResolvedEntitlementListResponse {
    pub fn new(data: Vec<ResolvedEntitlement>) -> Self {
        Self { data }
    }
}

File: typescript/src/models/resolvedEntitlementListResponse.ts
// this file is @generated
import {
  type ResolvedEntitlement,
  ResolvedEntitlementSerializer,
} from "./resolvedEntitlement";

export interface ResolvedEntitlementListResponse {
  data: ResolvedEntitlement[];
}

export const ResolvedEntitlementListResponseSerializer = {
  _fromJsonObject(object: any): ResolvedEntitlementListResponse {
    return {
      data: object["data"].map((item: any) =>
        ResolvedEntitlementSerializer._fromJsonObject(item)
      ),
    };
  },

  _toJsonObject(self: ResolvedEntitlementListResponse): any {
    return {
      data: self.data.map((item: any) =>
        ResolvedEntitlementSerializer._toJsonObject(item)
      ),
    };
  },
};
